### ISSUE TITLE
Make Terrain Brush "full tile" painting easier to find

### ISSUE THREAD

**Issue by bjorn** [mapeditor/tiled#3082]
The Terrain Brush can fill full tiles by holding Control. It is mentioned in the manual, but it would be nice if this feature was easier to discover. Also some people may want this mode to be the default, so that they don't have to hold Control all the time.

I'd suggest we add a toggle button, or two buttons to switch between, to the toolbar for the Terrain Brush. It should allow changing the current mode persistently, and Control can remain for temporarily switching to the other mode.

**Comment by eishiya** [comment]
If a toggle button or mode selector is added, I think the old full-tile behaviour should be the default, for consistency with older versions of Tiled. The toggle would make it easy to get the corner/edge-modifying behaviour for those that need it, but full-tile as the default would mean people coming from old versions of Tiled or following old tutorials wouldn't be confused.

**Comment by dnc77** [comment]
Hi there,

I decided to look at this project and consider this as a first fix. After some reviews on my end, I created a drop down and an option to enable/disable full tile mode. It definitely needs to be reviewed as I am still not 100% familiar with the project but it will look like the attached. Would you be happy to review this? Is the process right; to fork the repo, put in the change and commit it in the fork if you are happy to review this or is there another process please?

![change](https://user-images.githubusercontent.com/44986862/225291661-41735ed8-ee2f-4fc2-bff9-da7064b55622.png)

Cheers

Duncan

**Comment by eishiya** [comment]
This should not be a dropdown, but a tool option, similar to the random/terrain options on the Bucket Fill tool. There was also an icon for the toolbar already created on Discord, @bjorn presumably has it.

**Comment by dnc77** [comment]
Ah, ok - no it's certainly not my intention to try and find a better way - my apologies I like this project and decided to have a look and since I'm still not 100% on the way git supports collaboration I simply did not realize work has already been done on this. I am familiar with the concept of a branch where changes are made and they get approved and merged back to the main trunk and I am still accustoming myself to pull requests, forks and how they work.

While I spent time on this, I could say that I familiarized myself a little with this project and since an icon already exists, the feature already exists and is being considered for merging to the main trunk, I would gather that no further effort is required on this - would that be correct?


Thanks

Duncan

**Comment by eishiya** [comment]
GitHub doesn't distinguish between Issues and PRs very clearly in the issue feed (the only difference is the PR icon in the status indicator, which is easy to miss), but you can see the PR linked right above your first post in this thread.

As far as I can tell from the PR thread, it's done, and no further work is required besides implementing the icon, which should just be a matter of including the file and setting the icon path. Perhaps comparing your code to the code in the PR may be educational though, to see if you missed something, or did something another way.

**Comment by dnc77** [comment]
Yes - thanks - I'll do that as a next step. I know what you mean also regarding the tool options which get loaded on a second toolbar when the tool is selected. I came across that functionality while I was working through this issue as well but what you mentioned solidified better my understanding in that area. I'll see how this was done to compare with my understanding. Thanks a lot!

**Comment by bjorn** [comment]
@dnc77 Indeed the change done in #3407 would be preferable, but thanks a lot for handling this problem anyway! If you find anything else you'd like to help improve, please don't hesitate to get in touch. :-)

Also, I think your approach to grouping the tools could be interesting, since some tools could benefit from this. For example, I think the "shape fill tool" would be nicer when separated into "rectangle fill" and "circle fill" tools that would be selected by such a dropdown, as opposed to having a toggle option for that after selecting the tool. Similarly, a "circle select" tool could be added grouped with the "rectangular select" tool.

**Comment by bjorn** [comment]
Closing this since #3407 was merged eventually.
